### PR TITLE
Make codemirror work at runtime with Dart 2.0.

### PR DESCRIPTION
**Pull request by jcollins-g**
Tested with DartPad and passes unit tests, now.

### CODE REVIEW

**Comment by devoncarew** [comment]
Thanks for the separate commits! Looking now -

File: lib/codemirror.dart [hunk 1]
```diff
@@ -738,7 +738,7 @@ class Doc extends ProxyHolder {
    * content of the selections.
    */
   Iterable<String> getSelections([String lineSep]) =>
-      callArg('getSelections', lineSep);
+      new List<String>.from(callArg('getSelections', lineSep));
```
Comment (devoncarew) [inline lib/codemirror.dart]: I assume callArg() is not an iterable, so you're not using `.cast<String>()` here?

File: lib/src/js_utils.dart [hunk 2]
```diff
@@ -49,7 +49,7 @@ class JsEventListener {
 
   JsEventListener(this._proxy, this._name, {this.cvtEvent, this.twoArgs: false});
 
-  Stream get stream {
+  Stream<T> get stream {
```
Comment (devoncarew) [inline lib/src/js_utils.dart]: You could probably parameterize the `StreamController _controller` field as well.

File: lib/codemirror.dart (hunk 1 above)
Comment (jcollins-g) [reply lib/codemirror.dart]: nvm, I had thought it wasn't an iterable but it is.  Fixed.

File: lib/src/js_utils.dart (hunk 2 above)
Comment (jcollins-g) [reply lib/src/js_utils.dart]: done

**Pull request merged**